### PR TITLE
[Docs] Mention that datamodules can also be used with `.test()` method

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes https://github.com/PyTorchLightning/pytorch-lightning/issues/5265
Adds to docs that datamodules are also supported as a valid way to pass in an external dataloader

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes [if needed]?
- [ ] Did you write any new necessary tests [no need for typos, docs]? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [ ] **Check that target branch and milestone are aligned!**
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/test_set.rst [hunk 1]
```diff
@@ -3,6 +3,10 @@
 Test set
 ========
 Lightning forces the user to run the test set separately to make sure it isn't evaluated by mistake.
+Testing is preformed using the `trainer` objects `.test()` method.
+
+.. automethod:: pytorch_lightning.trainer.trainer.Trainer.test
```
Comment (rohitgr7) [inline docs/source/test_set.rst:8]:
```suggestion
Testing is performed using the ``trainer`` object's ``.test()`` method.

.. automethod:: pytorch_lightning.trainer.Trainer.test
```

File: docs/source/test_set.rst [hunk 2]
```diff
@@ -82,4 +86,19 @@ is not available at the time your model was declared.
     trainer.test(test_dataloaders=test)
 
 You can either pass in a single dataloader or a list of them. This optional named
-parameter can be used in conjunction with any of the above use cases.
+parameter can be used in conjunction with any of the above use cases. Additionally,
+you can also pass in an :ref:`datamodules` that have overridden the
+`test_dataloader` method.
```
Comment (rohitgr7) [inline docs/source/test_set.rst:91]:
```suggestion
``test_dataloader`` method.
```

Comment (rohitgr7) [reply docs/source/test_set.rst:91]: is there a way to link the docs for test_dataloader from the datamodule docs here? https://pytorch-lightning.readthedocs.io/en/latest/datamodules.html#test-dataloader

**Review by rohitgr7 — APPROVED** [review]
just minor comments

**Review by s-rog — APPROVED** [review]
Just gotta address rohitgr7's comments!

File: docs/source/test_set.rst (hunk 2 above)
Comment (SkafteNicki) [reply docs/source/test_set.rst]: @rohitgr7 I have added a label to the datamodule docs, which is now linked to

**Review by Borda — APPROVED** [review]
